pulse-metrics/src/pipeline/inflow/prom_scrape/http_sd.rs

```diff
@@ -23,10 +24,10 @@ pub struct TargetBlock {
 async fn do_fetch(config: &HttpServiceDiscovery) -> anyhow::Result<Vec<TargetBlock>> {
   log::debug!(
     "Fetching HTTP service discovery targets from {}",
-    config.url
+    env_or_inline_to_string(&config.url).unwrap_or_default()
```

Comment:
I would just fail if this returns no value given that it's core configuration. Something like:

```
env_or_inline_to_string(&config.url).ok_or_else(|| anyhow::Error(...))?
```